example/3d.js

```diff
@@ -0,0 +1,359 @@
+const glsl = require('glslify');
+const ndarray = require('ndarray');
+const pack = require('ndarray-pack');
+const unpack = require('ndarray-unpack');
+const nurbs = require('../nurbs');
+const wireframe = require('screen-projected-lines');
+const controlPanel = require('control-panel');
+const cross = require('gl-vec3/cross');
+const normalize = require('gl-vec3/normalize');
+
+const codeDiv = document.createElement('div');
+const codeLabel = document.createElement('button');
+codeLabel.textContent = 'Show/hide Code';
+const code = document.createElement('pre');
+codeDiv.append(codeLabel);
+codeDiv.append(code);
+document.body.appendChild(codeDiv);
+code.style.width = '600px';
+code.style.height = '60%';
+code.style.display = 'none';
+codeDiv.style.maxHeight = '60%';
+codeDiv.style.position = 'absolute';
+codeDiv.style.right = 0;
+codeDiv.style.bottom = 0;
+codeDiv.style.zIndex = 10;
+codeDiv.style.backgroundColor = 'white';
+codeDiv.style.padding = '10px';
+codeDiv.style.overflow = 'auto';
+codeDiv.style.fontSize = '9px';
+
+codeLabel.addEventListener('click', function () {
+  code.style.display = code.style.display === 'none' ? 'block' : 'none';
+});
+
+var error = document.createElement('span');
+error.style.color = '#cc3333';
+error.style.fontFamily = 'sans-serif';
+error.style.fontStyle = 'italic';
+error.style.position = 'absolute';
+error.style.left = '10px';
+error.style.bottom = '10px';
+error.style.zIndex = '1';
```

Comment:
It might have been cleaner to define all of these elements via `hyperscript` and use CSS classes for styling

```js
let elementsByName = {};
let s = function (name, el) {
  elementsByName[name] = el;
};
let containerEl = h('div', [
  s('codeLabel', h('button', 'Show/hide code'))
  s('code', h('pre'))
]);
// ...
```